common/src/main/java/discord4j/common/store/legacy/LegacyStoreLayout.java

```diff
@@ -354,16 +365,14 @@ public Mono<Void> onChannelCreate(int shardIndex, ChannelCreate dispatch) {
             case GUILD_CATEGORY:
             case GUILD_NEWS:
             case GUILD_STORE:
-            case GUILD_STAGE_VOICE: return saveChannel(dispatch);
+            case GUILD_STAGE_VOICE: return saveChannel(dispatch.channel());
```

Comment:
For this method and the two below, you should add the two following fields to the inner enum "Type" (L. ~440) and add them in the switch
```Java
PUBLIC_THREAD(11),
PRIVATE_THREAD(12),
```

Reply:
I will add new types to enum, but Channels with these types can come in the `ChannelCreate`?

Reply:
MB, I misread where I commented, I thought we were in the "saveChannel" method, sorry 😅 
In that case, no, but on the onThreadCreateMethod, you'll need to change the switch